### DISCUSSION TITLE
Question: why is preferred working group size set to number of cores, not number of threads ?

### DISCUSSION THREAD

**Discussion by boxerab** [pocl/pocl#1728] [Q&A]
I have an 8 core / 16 thread CPU.

from clinfo

```
 Preferred work group size multiple (device)     8
  Preferred work group size multiple (kernel)     8
```

Why 8 and not 16 ?

**Comment by pjaaskel** [comment]
This is not the work group *count* but work group *size* multiple. Work-groups execute in different threads in parallel, whereas work-items inside work-groups execute in one CPU thread and (hopefully) are autovectorized so that WIs get spread to vector lanes, thus the multiple should reflect the vector lane count.

**Comment by boxerab** [comment]
thanks, makes sense!
